### ISSUE TITLE
test_httplib2.test_https_success depends on side effects of other tests

### ISSUE THREAD

**Issue by sashahart** [cdent/wsgi-intercept#9]
I found this while trying out some restructuring of the tests to get rid of the globally shared state between different tests, and found that doing that caused this one test to start failing on python2. Other tests seemed to be working.

To reproduce, suppose I disable every other test in test_httplib2: for example, if I go into test_httplib2.py and change 'def test_success' to 'def _test_success' and 'def test_bogus_domain' to 'def _test_bogus_domain'. I see that test_https_success fails on python2 (python2.6, python2.7, pypy's 2.7) with ServerNotFoundError: Unable to find the server at some_hopefully_nonexistant_domain. Still seems OK on python3.

Re-enable either of the other tests, and it succeeds for all the versions.
As expected, same thing if you use py.test -k https_success to only run that one test without running the others in test_httplib2.

It stands to reason that this is because of the calls to install() affecting global state to install the intercept, and that state not being reset between tests, creating the implicit dependency of test_https_success on the others' installation of the intercept, and the failure mode of actually trying to do something with the hostname.

It depends on install() running at all - with all tests running, test_https_success fails in python2 depending on whether or not another test has called install() - tested by commenting out those lines.

It also depends on what argument is passed to install(). Again with all tests but test_https_success disabled, I modify test_https_success so that install(443) says instead install(). And I see the same test fails on python3 (python3.3, python3.4) with httplib2.ServerNotFoundError: Unable to find the server at some_hopefully_nonexistant_domain. Still seems OK on python2.

To summarize: in isolation from the other tests, calling install() lets test_https_success pass on python2 but fail on python3, while calling install(443) lets it pass on python3 but fail on python2. The question is why the intercept installation is failing to take in either case, almost certainly involving some python2 vs. python3 branch.

To be continued with more info
